### ISSUE TITLE
NoMethodError when building an exception for Faraday request

### ISSUE THREAD

**Issue by palkan** [palkan/isolator#21]
```
NoMethodError: undefined method` []' for #<Sniffer::DataItem::Request:0x00007fe477c8c040>

/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/isolator-0.6.0/lib/isolator/adapters/http/sniffer.rb:13:in `block in <top (required)>'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/isolator-0.6.0/lib/isolator/adapters/base.rb:50:in `build_exception'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/isolator-0.6.0/lib/isolator/adapters/base.rb:27:in `notify'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/isolator-0.6.0/lib/isolator/adapter_builder.rb:25:in `block (2 levels) in add_patch_method'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/sniffer-0.3.2/lib/sniffer/adapters/net_http_adapter.rb:28:in `request_with_sniffer'
/.rbenv/versions/2.5.1/lib/ruby/2.5.0/net/http.rb:1457:in `block in request'
/.rbenv/versions/2.5.1/lib/ruby/2.5.0/net/http.rb:910:in `start'
/.rbenv/versions/2.5.1/lib/ruby/2.5.0/net/http.rb:1455:in `request'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/sniffer-0.3.2/lib/sniffer/adapters/net_http_adapter.rb:32:in `block in request_with_sniffer'
/.rbenv/versions/2.5.1/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/sniffer-0.3.2/lib/sniffer/adapters/net_http_adapter.rb:31:in `request_with_sniffer'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/faraday-0.15.2/lib/faraday/adapter/net_http.rb:87:in `perform_request'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/faraday-0.15.2/lib/faraday/adapter/net_http.rb:43:in `block in call'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/faraday-0.15.2/lib/faraday/adapter/net_http.rb:92:in `with_net_http_connection'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/faraday-0.15.2/lib/faraday/adapter/net_http.rb:38:in `call'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/faraday-0.15.2/lib/faraday/rack_builder.rb:143:in `build_response'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/faraday-0.15.2/lib/faraday/connection.rb:387:in `run_request'
/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/faraday-0.15.2/lib/faraday/connection.rb:175:in `put'
/app/services/fake_ssn_uploader_service.rb:15:in `call'```
```

Failing code:

```ruby
class FakeSsnUploaderService
  attr_reader :url, :value

  def initialize(url, value)
    @url = url
    @value = value
  end

  def call
    connection.put(url, value) do |f|
      f.headers["Content-Type"] = "text/plain"
    end
  end

  private

  def connection
    Faraday.new do |faraday|
      # faraday.use Faraday::Response::RaiseError
      faraday.adapter Faraday.default_adapter
    end
  end
end

ActiveRecord::Base.transaction do
  FakeSsnUploaderService.new(url, data).call
end
```

**Comment by steobrien** [comment]
Any workaround for this issue that you could recommend?

**Comment by palkan** [comment]
Fixed in https://github.com/palkan/isolator/commit/f732227bdb11489bdf4dea41e4f53fe929d2bf2f and released in 0.6.1.

**Comment by steobrien** [comment]
Awesome! Thank you.
